src/providers/on-chain-quote-provider.ts

```diff
@@ -332,20 +341,23 @@ export class OnChainQuoteProvider implements IOnChainQuoteProvider {
     );
   }
 
-  private async getQuotesManyData(
+  private async getQuotesManyData<
+    TRoute extends V3Route | V2Route | MixedRoute
+  >(
     amounts: CurrencyAmount[],
-    routes: (V2Route | V3Route | MixedRoute)[],
+    routes: TRoute[],
     functionName: 'quoteExactInput' | 'quoteExactOutput',
     _providerConfig?: ProviderConfig
   ): Promise<{
-    routesWithQuotes: RouteWithQuotes[];
+    routesWithQuotes: RouteWithQuotes<TRoute>[];
     blockNumber: BigNumber;
   }> {
-    const isMixedRoutes = routes.every(
-      (route) => route.protocol === Protocol.MIXED
-    );
+    const usesMixedRouteQuoter =
+      routes.some((route) => route instanceof MixedRoute) ||
+      routes.some((route) => route instanceof V2Route);
 
-    this.validateRoutes(routes, functionName, isMixedRoutes);
+    /// Validate that there are no incorrect routes / funcitonName combinations
+    this.validateRoutes(routes, functionName, usesMixedRouteQuoter);
```

Comment:
While we will never make a call to `getQuotesManyExactIn()` with a mixture of V3Route | V2Route | MixedRoute through AlphaRouter (since we only use routes created by functions like `computeAllV3Routes`, etc., since this class can be exported we should prevent errors that can arise from incorrect usages.